fix: correct WorkflowResult error type from Error to string

RichardLoRicco opened the pull request:
Fixes issue #5848 where the `WorkflowResult` type declared `error` as an `Error` object, but the workflow system actually returns error messages as strings at runtime.

**Primary fix:** Updated `WorkflowResult` interface to correctly type the error field as `string` instead of `Error`.

**Secondary fix:** While investigating, discovered the agent-builder was also making incorrect assumptions about the error type. Added a type guard to safely handle both strings and Error objects.

**Before:**
```typescript
// Type definition - incorrect
interface WorkflowResult {
  error: Error;  // Runtime value is actually string
}

// Agent builder transformWorkflowResult - assumes Error object
} else if (result.status === 'failed') {
  return {
    success: false,
    applied: false,
    message: `Agent builder action failed: ${result.error.message}`,  // Would throw on string
    error: result.error.message,
  };
}
```

After:
```typescript
// Type definition - corrected
interface WorkflowResult {
  error: string;  // Matches runtime behavior
}

// Agent builder transformWorkflowResult - handles both safely
} else if (result.status === 'failed') {
  const errorMessage = typeof result.error === 'string' ? result.error : result.error.message;
  return {
    success: false,
    applied: false,
    message: `Agent builder action failed: ${errorMessage}`,
    error: errorMessage,
  };
}
```

Added tests to verify the corrected behavior.

Co-Authored-By: Harold Camacho <harold.camacho.diaz@gmail.com>

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling consistency by ensuring all errors are properly serialized and typed as strings over HTTP, replacing variable error object formats.

* **Tests**
  * Added test case to verify error messages are consistently handled as strings in workflow results.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

RichardLoRicco committed:
test: add test for issue #5848 - verify workflow error is string not Error

RichardLoRicco committed:
fix: correct WorkflowResult error type from Error to string

Updates WorkflowResult type definition to match actual runtime behavior where errors are serialized as strings. Adds test with compile-time and runtime type verification.

RichardLoRicco committed:
fix: handle string errors in agent-builder transformWorkflowResult

Add type guard to support both string and Error object types. Add tests for both cases.

DanielSLew commented on client-sdks/client-js/src/resources/agent-builder.test.ts:
```diff
@@ -130,3 +130,63 @@ describe('AgentBuilder.runs', () => {
   });
 });
+
+describe('AgentBuilder.transformWorkflowResult', () => {
```
I don't think we need these tests, I think the workflow test is enough that we're passing through the correct type

DanielSLew reviewed (approved):
Just a small nit but looks good

RichardLoRicco committed:
chore: remove redundant AgentBuilder.transformWorkflowResult tests

Remove the AgentBuilder.transformWorkflowResult test suite as the workflow tests provide sufficient coverage for the error type correction.

DanielSLew committed:
Merge branch 'main' into fix/workflow-error-type-correction
